core/test/nuxeo-audit-page-provider.test.js

```diff
@@ -0,0 +1,231 @@
+      const event = await waitForEvent(provider, 'notify');
+      expect(event).to.have.property('detail');
+      expect(event.detail).to.have.property('error', error);
+    });
+
+    test('Should return an error when occurs in the Rest call', async () => {
+      mockDocHistoryResponse(error);
+      const provider = await getProvider(providerParameters);
+      try {
+        await provider.fetch();
+        return Promise.reject(new Error('Should have thrown an error'));
+      } catch (err) {
+        expect(err).to.be.an('Error');
+        expect(err).to.have.property('message', error.message);
+      }
+    });
+
+    test('Should use Rest call with sorting when document identifier is provided', async () => {
+      mockDocHistoryResponse();
+      const provider = await getProvider(providerParameters, documentId, providerSort);
+      const qParams = {
+        sortBy: 'eventDate,category,eventId,principalName',
+        sortOrder: 'asc,desc,desc,asc',
+        ...requestParameters,
+      };
+      checkResponse(await provider.fetch());
+      assertRequestQueryParams(qParams);
+    });
+  });
+
+  suite('Fetch Audit', () => {
+    test('Should use Operation call when document identifier is not provided', async () => {
+      mockAuditResponse();
+      const provider = await getProvider(providerParameters, '');
+      checkResponse(await provider.fetch());
+      assertRequestBodyParams(providerParameters);
+    });
+
+    test('Should fire a success event using Operation call when document identifier is not provided', async () => {
+      mockAuditResponse();
+      const provider = await getProvider(providerParameters, '');
+      provider.fetch();
+      const event = await waitForEvent(provider, 'update');
+      expect(event).to.have.property('detail');
+      expect(event.detail).to.be.null;
+    });
+
+    test('Should fire an event when an error occurs in the Operation call', async () => {
+      // 'nuxeo-operation' needs the response to be present even in error case
+      error.response = {};
+      mockAuditResponse(error);
+      const provider = await getProvider(providerParameters, '');
+      provider.fetch();
+      const event = await waitForEvent(provider, 'notify');
+      expect(event).to.have.property('detail');
+      expect(event.detail).to.have.property('error', error);
+      error.response = null;
+    });
+
+    test('Should return an error when occurs in the Operation call', async () => {
+      // 'nuxeo-operation' needs the response to be present even in error case
+      error.response = {};
+      mockAuditResponse(error);
+      const provider = await getProvider(providerParameters, '');
+      try {
+        await provider.fetch();
+        return Promise.reject(new Error('Should have thrown an error'));
+      } catch (err) {
+        expect(err).to.be.an('Error');
+        expect(err).to.have.property('message', error.message);
+      }
+      error.response = null;
+    });
```

Comment:
nitpick: I'm more fan of not sharing lot of stuff between tests, but I understand the factoring. 
In this case we end up with a `setup` and `teardown` for the `error` variable inside the test because we are sharing it between all the tests.

Reply:
yes, I know and I agree, `setup` and `before` are the best place of those things, but here we don't have the same behaviour on `nuxeo-operation` and `nuxeo-resource` in the way that they handle the error. it's the main reason. at least the two elements should check if error is not `null` and check `error.response` if needed. which is not the case. 

to go deeper, currently the `mock-client` accept `null` as an error. But if we use it it will fails on the same elements: `nuxeo-operation` and `nuxeo-resource`